Source/NSButton.m

```diff
@@ -543,4 +547,43 @@ - (NSSound *) sound
   return [_cell sound];
 }
 
+- (void) mouseDown: (NSEvent *)event
+{
+  [super mouseDown: event];
+
+  if ([[self superview] isKindOfClass: [NSMatrix class]] == NO)
```

Comment:
Did you test, whether the exclusion of NSMatrix is enforced on Cocoa? And if so, which other behaviour is expected there. Do we actually implement this behaviour?

Reply:
The NSMatrix class is deprecated on Cocoa.   I added this to avoid any potential conflicts with existing logic in GS.

Reply:
You don't need to worry about that. The logic in NSMatix is based on cells not subviews. Normally an NSMatrix won't have subviews at all.